### PR TITLE
e2e: ingress class

### PR DESCRIPTION
**Pull request by akshaysngupta**
## Checklist
- [x] The title of the PR is clear and informative
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

This PR adds an e2e for ingress class.
We deploy AGIC again with ingress class set to "custom-ingress-class" and check that ingress with this class are picked.

### CODE REVIEW

File: scripts/e2e/cmd/runner/helper.go [hunk 1]
```diff
@@ -397,3 +434,45 @@ func makeGetRequest(url string, host string, statusCode int, inSecure bool) (*ht
 
 	return nil, fmt.Errorf("Unable to get status code %d with url '%s' with host '%s'. Response: [%+v]", statusCode, url, host, resp)
 }
+
+func validateSubResourcePrefix() error {
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/helper.go:438]: I think the resourcePrefix is already covered in your previous PR already? better to do in different PR

Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/helper.go:438]: I forgot to mention that this PR is based on the previous PR, so will remove changes

File: scripts/e2e/run-e2e.sh [hunk 2]
```diff
@@ -16,3 +16,12 @@ go test -v -timeout 60m -tags e2e ./... >testoutput.txt || {
     cat testoutput.txt
     exit 1
 }
+
+# install with custom tag
+InstallAGIC "custom-ingress-class"
+
+go test -v -timeout 60m -tags e2eingressclass ./... >testoutput.txt || {
```
Comment (3quanfeng) [inline scripts/e2e/run-e2e.sh:23]: -tags "e2e e2eingressclass"?

File: scripts/e2e/cmd/runner/ingress_class_test.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+			time.Sleep(30 * time.Second)
+
+			// get ip address for 1 ingress
+			klog.Info("Getting public IP from Ingress...")
+			publicIP, _ := getPublicIP(clientset, namespaceName)
+			Expect(publicIP).ToNot(Equal(""))
+
+			url := fmt.Sprintf("http://%s/status/200", publicIP)
+
+			// should return 404 as it should not be plumbed
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/ingress_class_test.go:61]: we probably need to cover the valid case for both AGICs with different kubernetes.io/ingress.class, customer is gonna use this feature to manage different services within same namespaces but different AGICs

File: scripts/e2e/run-e2e.sh (hunk 2 above)
Comment (akshaysngupta) [reply scripts/e2e/run-e2e.sh]: no, e2eingressclass is enough

File: scripts/e2e/cmd/runner/ingress_class_test.go (hunk 3 above)
Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/ingress_class_test.go:61]: in this test, we use two ingress yamls.
1 with our standard ingress class
and 1 with custom ingress class.

we expect that AGIC picks up custom ingress class.

File: scripts/e2e/cmd/runner/ingress_class_test.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
+// -------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// --------------------------------------------------------------------------------------------
+
+// +build e2e e2eingressclass
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/ingress_class_test.go:6]: remove e2e tag here?

File: scripts/e2e/run-e2e.sh (hunk 2 above)
Comment (3quanfeng) [reply scripts/e2e/run-e2e.sh]: didn't see your change about separating those two

**Review by 3quanfeng — COMMENTED** [review]
LGTM!

**Review by 3quanfeng — APPROVED** [review]
